pepper_gazebo_plugin/scripts/laser_publisher.py

```diff
@@ -16,117 +17,91 @@ def __init__(self):
             rospy.init_node('laser_test', anonymous=True, disable_signals=True)
             print "Initialised rospy node: laser_test"
 
-        self.lp = LaserProjection()
         self.tl = TransformListener()
+        self.lp = LaserProjection()
 
-        self.left_laser_scans = []
-        self.right_laser_scans = []
-        self.front_laser_scans = []
+        self.all_laser_pub = rospy.Publisher(
+            '/pepper/laser', LaserScan, queue_size=1)
 
+        # Subscribers
         left_sub = Subscriber('/pepper/scan_left', LaserScan)
         front_sub = Subscriber('/pepper/scan_front', LaserScan)
         right_sub = Subscriber('/pepper/scan_right', LaserScan)
+
         self.ts = TimeSynchronizer([left_sub, front_sub, right_sub],
                                    10)
-        self.ts.registerCallback(self.publish_all_lasers)
+        self.ts.registerCallback(self.scan_cb)
 
-        self.all_laser_pub = rospy.Publisher(
-            '/pepper/laser', LaserScan, queue_size=1)
         print "Finished intialising"
 
-    # def laser_scan_cb(self, left, front, right):
-    #     self.left_laser_scans.append(left)
-    #     self.right_laser_scans.append(right)
-    #     self.front_laser_scans.append(front)
+    def scan_cb(self, left, front, right):
+        translated_points = []
+        try:
+            pc_left = self.lp.projectLaser(left)
+            pc_front = self.lp.projectLaser(front)
+            pc_right = self.lp.projectLaser(right)
+        except Exception as e:
+            print "Failed to transform laser scan because: " + str(e)
```

Comment:
Use prints in the Python3 format please `print("bla")`, or even better, given it's a ROS node, use `rospy.logerr`.